packages/asset/contracts/AssetCreate.sol

```diff
@@ -354,5 +355,197 @@ contract AssetCreate is
         return ERC2771HandlerUpgradeable._msgData();
     }
 
-    uint256[45] private __gap;
+    bytes32 public constant LAZY_MINT_TYPEHASH =
+        keccak256(
+            "LazyMint(address creator,uint16 nonce,uint8 tier,uint256 amount,string metadataHash,uint256 maxSupply)"
+        );
+
+    bytes32 public constant LAZY_MINT_BATCH_TYPEHASH =
+        keccak256(
+            "LazyMintBatch(address[] creators,uint16 nonce,uint8[] tiers,uint256[] amounts,string[] metadataHashes,uint256[] maxSupplies)"
+        );
+
+    // Address of the vault that distributes the catalyst value to the creators
+    ILazyVault public vault;
+
+    // mapping of tokenId => maxSupply specified by the creator
+    mapping(uint256 => uint256) public availableToMint;
+
+    /// @notice Lazily creates a new asset with a signature
+    /// @param signature A signature generated by TSB
+    /// @param tier The tier of the asset to mint
+    /// @param amount The amount of the asset to mint
+    /// @param metadataHash The metadata hash of the asset to mint
+    /// @param maxSupply The max supply of the asset to mint
+    /// @param creator The address of the creator
+    function lazyCreateAsset(
+        bytes memory signature,
+        uint8 tier,
+        uint256 amount,
+        string calldata metadataHash,
+        uint256 maxSupply,
+        address creator
+    ) external whenNotPaused {
```

Comment:
When I look at `createAsset` and `lazyCreateAsset` its not obvious about which parts to merge?
If we removed maxSupply we could probably combine those two functions together because the signature would be the same. We would just need to check if sender != creator address.